pkg/informer/runInformers.go

```diff
@@ -13,33 +19,437 @@ import (
+	resource := strings.TrimSuffix(crd.GetName(), "."+group)
+	if resource == "" {
+		return nil, fmt.Errorf("failed to parse the resource name from the CRD: %s", crd.GetName())
+	}
+
+	return &schema.GroupVersionResource{
+		Group:    group,
+		Version:  versionName,
+		Resource: resource,
+	}, nil
+}
+
+// gvrToPrinterColumns is a concurrency safe mapping of the stored version of a CRD represented as a
+// schema.GroupVersionResource with values of ExtractProperty slices which represent the additionalPrinterColumns.
+type gvrToPrinterColumns struct {
+	lock    sync.RWMutex
+	mapping map[schema.GroupVersionResource][]tr.ExtractProperty
+}
+
+// toLowerCamelCase converts a UTF-8 string to lower camel case such as enforcementAction. Invalid UTF-8 words are
+// ignored. The considered separators are spaces (e.g. "enforcement action") and dashes (e.g. enforcement-action).
+func toLowerCamelCase(s string) string {
+	if s == "" {
+		return s
+	}
+
+	s = strings.TrimSpace(s)
+	s = strings.ToLower(s)
+
+	for _, separator := range []string{" ", "-"} {
+		parts := strings.Split(s, separator)
+
+		newS := ""
+
+		for i, part := range parts {
+			if i != 0 {
+				r, size := utf8.DecodeRuneInString(part)
+				// RuneError should never be returned because Kubernetes should ensure it's UTF-8 characters, but this is
+				// just an extra precaution.
+				if r == utf8.RuneError {
+					continue
+				}
+
+				part = string(unicode.ToUpper(r)) + part[size:]
+			}
+
+			newS += part
+		}
+
+		s = newS
+	}
+
+	return s
+}
+
+// set will parse the stored GVR from the input CRD and set it in the mapping cache.
+func (g *gvrToPrinterColumns) set(crd *unstructured.Unstructured) error {
+	gvr, err := gvrFromCRD(crd)
+	if err != nil {
+		return err
+	}
+
+	var printerColumns []tr.ExtractProperty
+
+	versions, _, _ := unstructured.NestedSlice(crd.Object, "spec", "versions")
+	for _, version := range versions {
+		versionTyped, ok := version.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		name, _ := versionTyped["name"].(string)
+		if name != gvr.Version {
+			continue
+		}
+
+		additionalPrinterColumns, ok := versionTyped["additionalPrinterColumns"].([]interface{})
+		if !ok {
+			break
+		}
+
+		for _, column := range additionalPrinterColumns {
+			columnTyped, ok := column.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			name, ok := columnTyped["name"].(string)
+			if !ok {
+				continue
+			}
+
+			jsonPath, ok := columnTyped["jsonPath"].(string)
+			if !ok {
+				continue
+			}
+
+			// The additionalPrinterColumns always have a JSON path without curly braces enclosing it, but the
+			// ExtractProperty.JSONPath field expects them.
+			jsonPath = fmt.Sprintf("{%s}", jsonPath)
+
+			camelCaseName := toLowerCamelCase(name)
+
+			if camelCaseName == "" {
+				continue
+			}
+
+			printerColumns = append(printerColumns, tr.ExtractProperty{Name: camelCaseName, JSONPath: jsonPath})
```

Comment:
Can you add `Type` (string, integer, etc) to ExtractProperty{} ?
Then, we'll need to use the `Type` later when transforming the resource.

Reply:
@jlpadilla how about changing the `GenericResourceBuilder` assignment to leverage the underlying type found by Go? We know it'll be limited to valid JSON properties since the Kubernetes API returned the object as JSON before it became a `map[string]interface{}` object when unmarshaled?

```go
n.Properties[prop.Name] = fmt.Sprintf("%s", result[0][0])
```

To:

```go
n.Properties[prop.Name] = result[0][0].Interface()
```

Reply:
I ended up going with the approach I mentioned in my previous comment.